File: BUFF/BuffServer/src/main/java/blserviceimpl/strategy/BackData.java
package blserviceimpl.strategy;

/**
 * Created by wshwbluebird on 2017/3/27.
 */

import stockenum.StockPickIndex;
import stockenum.StrategyType;

/**
 * 用来封装要比较的数据类型
 * 减少文件的读如次数
 */
public class BackData {
    public String code;

    /**
     * 传进来用来比较数据的值
     */
    public Number rankValue;

    public Number[] mixRank;

    public Number[] filterData;

    /**
     * 第一天买入时的价格。对应第一天的开盘价
     */
    public double firstDayOpen;

    /**
     * 最后一天卖出时的价格，对应最后一天的收盘价
     */
    public double lastDayClose;

    /**
     * 不带过滤参数的初始化过滤方法
     * @param code
     * @param rankValue
     * @param firstDayOpen
     * @param lastDayClose
     */
    public BackData(String code , Number rankValue, double firstDayOpen, double lastDayClose){
        this.code = code;
        this.rankValue = rankValue;
        this.firstDayOpen = firstDayOpen;
        this.lastDayClose = lastDayClose;
        filterData  =new Number[StockPickIndex.values().length];
        mixRank = new Number[StrategyType.values().length];
    }

    /**
     * 带过滤参数的方法
     * @param code
     * @param rankValue
     * @param firstDayOpen
     * @param lastDayClose
     */
    public BackData(String code , Number rankValue, double firstDayOpen, double lastDayClose, Number[] filterData){
        this.code = code;
        this.rankValue = rankValue;
        this.firstDayOpen = firstDayOpen;
        this.lastDayClose = lastDayClose;
        this. filterData  = filterData;
    }

}

File: BuffTreasureWebApp/vo/BackData.js
/**
 * Created by slow_time on 2017/5/11.
 */
exports.BackData = class {
    /**
     * VO构造函数
     * @param code {String} 股票代码
     * @param name {String} 股票名称
     * @param rankValue {Number} 用于比较的值
     * @param mixRank {Array} 混合策略中各个策略的比较值构成的数组
     * @param filterData {Array} 过滤参数
     * @param firstDayOpen {Number} 第一天买入时的价格。对应第一天的开盘价
     * @param lastDayClose {Number} 最后一天卖出时的价格，对应最后一天的收盘价
     * @param valid {bool} 这几天是否有停盘  有的话 flase 没有就是 true
     */
    constructor(code, name, rankValue, mixRank, filterData, firstDayOpen, lastDayClose ,valid) {
        this.code = code;
        this.name = name;
        /**
         * 传进来用来比较数据的值
         */
        this.rankValue = rankValue;
        this.mixRank = mixRank;
        this.filterData = filterData;
        /**
         * 第一天买入时的价格。对应第一天的开盘价
         * adjust by wsw 也可能是前一天的复权收盘价
         */
        this.firstDayOpen = firstDayOpen;

        /**
         * 最后一天卖出时的价格，对应最后一天的收盘价
         */
        this.lastDayClose = lastDayClose;
        this.valid = valid;
    }
};